packages/hoprd/crates/hopr-lib/src/processes/mod.rs

```diff
@@ -1,77 +1,134 @@
     db: Arc<RwLock<Db>>,
     multi_strategy: Arc<MultiStrategy>,
-    channel_graph: Arc<RwLock<ChannelGraph>>,
-) -> futures::channel::mpsc::UnboundedSender<ChannelEntry>
+    mut event_stream: UnboundedReceiver<SignificantChainEvent>,
+    channel_graph: Arc<RwLock<core_path::channel_graph::ChannelGraph>>,
+    transport_indexer_actions: core_transport::IndexerActions,
+)
 where
     Db: core_ethereum_db::traits::HoprCoreEthereumDbActions + 'static,
 {
-    let (on_channel_event_tx, mut rx) = unbounded::<ChannelEntry>();
-
     spawn_local(async move {
-        while let Some(channel) = poll_fn(|cx| Pin::new(&mut rx).poll_next(cx)).await {
-            let maybe_direction = channel.direction(&me);
-            let change = channel_graph.write().await.update_channel(channel);
+        while let Some(event) = event_stream.next().await {
+            match event {
+                SignificantChainEvent::Announcement(peer, address, multiaddresses) => {
+                    if let Ok(peer) = PeerId::from_str(&peer) {
+                        if peer != me {
+                            // decapsulate the `p2p/<peer_id>` to remove duplicities
+                            let mas = multiaddresses
+                                .into_iter()
+                                .filter_map(|ma_str| core_transport::Multiaddr::from_str(&ma_str).ok())
+                                .map(|ma| core_transport::decapsulate_p2p_protocol(&ma))
+                                .filter(|v| !v.is_empty())
+                                .collect::<Vec<_>>();
+            
+                            if mas.len() > 0 {
+                                transport_indexer_actions
+                                    .emit_indexer_update(IndexerToProcess::Announce(peer.clone(), mas))
+                                    .await;
+            
+                                if db
+                                    .read()
+                                    .await
+                                    .is_allowed_to_access_network(&address)
+                                    .await
+                                    .unwrap_or(false)
+                                {
+                                    transport_indexer_actions
+                                        .emit_indexer_update(IndexerToProcess::EligibilityUpdate(peer, true.into()))
+                                        .await;
+                                }
+                            }
+                        } else {
+                            debug!("Skipping announcements for myself ({peer})");
+                        }
+                    } else {
+                        error!("Announced PeerId ({peer}) has invalid format")
+                    }
+                },
+                SignificantChainEvent::ChannelUpdate(channel) |
+                SignificantChainEvent::TicketRedeem(channel, _) => {
+                    let maybe_direction = channel.direction(&me_onchain);
```

Comment:
@NumberFour8 is the channel update logic correct here? Indexer -> channel graph

Reply:
I think the TicketRedeem is not needed, only `ChannelOpen, ChannelClose, ChannelCloseInitiated` are currently needed for the graph changes.